packages/cli/src/ui/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,455 @@
+        );
+      },
+    },
+    {
+      label: 'Respect .gitignore',
+      value: 'fileFiltering.respectGitIgnore',
+      checked:
+        getScopedNestedValue('fileFiltering', 'respectGitIgnore') ?? false,
+      toggle: () => {
+        updateFileFiltering(
+          'respectGitIgnore',
+          !(getScopedNestedValue('fileFiltering', 'respectGitIgnore') ?? false),
+        );
+      },
+    },
+    {
+      label: 'Respect .geminiignore',
+      value: 'fileFiltering.respectGeminiIgnore',
+      checked:
+        getScopedNestedValue('fileFiltering', 'respectGeminiIgnore') ?? false,
+      toggle: () => {
+        updateFileFiltering(
+          'respectGeminiIgnore',
+          !(
+            getScopedNestedValue('fileFiltering', 'respectGeminiIgnore') ??
+            false
+          ),
+        );
+      },
+    },
+    {
+      label: 'Enable Recursive File Search',
+      value: 'fileFiltering.enableRecursiveFileSearch',
+      checked:
+        getScopedNestedValue('fileFiltering', 'enableRecursiveFileSearch') ??
+        false,
+      toggle: () => {
+        updateFileFiltering(
+          'enableRecursiveFileSearch',
+          !(
+            getScopedNestedValue(
+              'fileFiltering',
+              'enableRecursiveFileSearch',
+            ) ?? false
+          ),
+        );
+      },
+    },
+    {
+      label: 'Hide Window Title',
+      value: 'hideWindowTitle',
+      checked: getScopedValue('hideWindowTitle') ?? false,
+      toggle: () => {
+        updateSetting(
+          'hideWindowTitle',
+          !(getScopedValue('hideWindowTitle') ?? false),
+        );
+      },
+    },
+    {
+      label: 'Hide Tips',
+      value: 'hideTips',
+      checked: getScopedValue('hideTips') ?? false,
+      toggle: () => {
+        updateSetting('hideTips', !(getScopedValue('hideTips') ?? false));
+      },
+    },
+    {
+      label: 'Hide Banner',
+      value: 'hideBanner',
+      checked: getScopedValue('hideBanner') ?? false,
+      toggle: () => {
+        updateSetting('hideBanner', !(getScopedValue('hideBanner') ?? false));
+      },
+    },
+    {
+      label: 'IDE Mode',
+      value: 'ideMode',
+      checked: getScopedValue('ideMode') ?? false,
+      toggle: () => {
+        updateSetting('ideMode', !(getScopedValue('ideMode') ?? false));
+      },
+    },
+    {
+      label: 'Vim Mode',
+      value: 'vimMode',
+      checked: getScopedValue('vimMode') ?? false,
+      toggle: () => {
+        updateSetting('vimMode', !(getScopedValue('vimMode') ?? false));
+      },
+    },
+    {
+      label: 'Disable Auto Update',
+      value: 'disableAutoUpdate',
+      checked: getScopedValue('disableAutoUpdate') ?? false,
+      toggle: () => {
+        updateSetting(
+          'disableAutoUpdate',
+          !(getScopedValue('disableAutoUpdate') ?? false),
+        );
+      },
+    },
+  ];
+
+  // Scope selector items
+  const scopeItems = [
```

Comment:
can you refactor to share code with the identical implementation in the themes dialog.